GPSTest/src/main/java/com/android/gpstest/util/UIUtils.java

```diff
@@ -291,17 +292,49 @@ public static void sendEmail(Context context, String email, String location) {
 
     /**
      * Returns the provided latitude or longitude value in Degrees Minutes Seconds (DMS) format
-     * @param latOrLon latitude or longitude to convert to DMS format
+     * @param coordinate latitude or longitude to convert to DMS format
      * @return the provided latitude or longitude value in Degrees Minutes Seconds (DMS) format
      */
-    public static String getDMSFromLocation(Context context, double latOrLon) {
-        BigDecimal loc = new BigDecimal(latOrLon);
+    public static String getDMSFromLocation(Context context, double coordinate, String latOrLon) {
+        BigDecimal loc = new BigDecimal(coordinate);
         BigDecimal degrees = loc.setScale(0, RoundingMode.DOWN);
         BigDecimal minTemp = loc.subtract(degrees).multiply((new BigDecimal(60))).abs();
         BigDecimal minutes = minTemp.setScale(0, RoundingMode.DOWN);
-        BigDecimal seconds = minTemp.subtract(minutes).multiply(new BigDecimal(60)).setScale(0, RoundingMode.DOWN);
+        BigDecimal seconds = minTemp.subtract(minutes).multiply(new BigDecimal(60)).setScale(2, RoundingMode.HALF_UP);
 
-        return context.getString(R.string.gps_lat_lon_dms_value, degrees.intValue(), minutes.intValue(), seconds.intValue());
+        String hemisphere;
+        int output_string;
+        if ( latOrLon.equals("lat") ) {
+            hemisphere = (coordinate < 0 ? "S" : "N");
```

Comment:
Do "N", "S", "E", and "W" need to be translated to other languages?  Or are they generally accepted as part of  [coordinate formats](https://en.wikipedia.org/wiki/Geographic_coordinate_conversion) in other languages?

@mirabilos @LLZN @SakuraSa233 @SIRSteiner @Skimige @D3nnisd Any thoughts on this?

If they need to be translated the strings should be moved to the main `strings.xml`, if not they should be moved to `do_not_translate.xml`

Reply:
In a word, I suggest that we do not translate them. I'll give my explanation if needed.